vertical-pod-autoscaler/pkg/utils/vpa/capping_test.go

```diff
@@ -184,3 +184,57 @@ func TestApply(t *testing.T) {
 		assert.Equal(t, testCase.ExpectedError, err)
 	}
 }
+
+func TestApplyVpa(t *testing.T) {
+	podRecommendation := vpa_types.RecommendedPodResources{
+		ContainerRecommendations: []vpa_types.RecommendedContainerResources{
+			{
+				ContainerName: "ctr-name",
+				Target: apiv1.ResourceList{
+					apiv1.ResourceCPU:    *resource.NewScaledQuantity(10, 1),
+					apiv1.ResourceMemory: *resource.NewScaledQuantity(5000, 1),
+				},
+				UncappedTarget: apiv1.ResourceList{
+					apiv1.ResourceCPU:    *resource.NewScaledQuantity(10, 1),
+					apiv1.ResourceMemory: *resource.NewScaledQuantity(5000, 1),
+				},
+				LowerBound: apiv1.ResourceList{
+					apiv1.ResourceCPU:    *resource.NewScaledQuantity(50, 1),
+					apiv1.ResourceMemory: *resource.NewScaledQuantity(4300, 1),
+				},
+			},
+		},
+	}
+	policy := vpa_types.PodResourcePolicy{
+		ContainerPolicies: []vpa_types.ContainerResourcePolicy{
+			{
+				ContainerName: "ctr-name",
+				MinAllowed: apiv1.ResourceList{
+					apiv1.ResourceCPU:    *resource.NewScaledQuantity(40, 1),
+					apiv1.ResourceMemory: *resource.NewScaledQuantity(4000, 1),
+				},
+				MaxAllowed: apiv1.ResourceList{
+					apiv1.ResourceCPU:    *resource.NewScaledQuantity(45, 1),
+					apiv1.ResourceMemory: *resource.NewScaledQuantity(4500, 1),
+				},
+			},
+		},
+	}
+
+	res, err := ApplyVPAPolicy(&podRecommendation, &policy)
+	assert.Nil(t, err)
+	assert.Equal(t, apiv1.ResourceList{
+		apiv1.ResourceCPU:    *resource.NewScaledQuantity(40, 1),
+		apiv1.ResourceMemory: *resource.NewScaledQuantity(4500, 1),
+	}, res.ContainerRecommendations[0].Target)
+
+	assert.Equal(t, apiv1.ResourceList{
+		apiv1.ResourceCPU:    *resource.NewScaledQuantity(10, 1),
+		apiv1.ResourceMemory: *resource.NewScaledQuantity(5000, 1),
+	}, res.ContainerRecommendations[0].UncappedTarget)
+
+	assert.Equal(t, apiv1.ResourceList{
+		apiv1.ResourceCPU:    *resource.NewScaledQuantity(45, 1),
```

Comment:
Do we really expect LowerBound > Target ?

Reply:
Fixed.